include/bio/seq_io/reader_options.hpp

```diff
@@ -185,9 +215,43 @@ struct reader_options
     //!\brief Truncate IDs at first whitespace.
     bool truncate_ids = false;
 
-    // TODO static_assert
-};
+private:
+    static_assert(detail::is_fields_tag<field_ids_t>, "field_ids must be a bio::vtag over bio::field.");
 
-//!\}
+    static_assert(detail::is_type_list<field_types_t>, "field_types must be a bio::ttag / seqan3::type_list.");
+
+    static_assert(detail::is_type_list<formats_t>, "formats must be a bio::ttag / seqan3::type_list.");
+
+    static_assert(field_ids_t::size == seqan3::list_traits::size<field_types_t>,
+                  "field_ids and field_types must have the same size.");
+
+    //!\brief Type of the record.
+    using record_t = detail::record_from_typelist<field_ids_t, field_types_t>;
+
+    static_assert(
+      detail::lazy_concept_checker([](auto) requires(
+        !field_ids_t::contains(field::id) ||
+        detail::back_insertable_with<record_element_t<field::id, record_t>, char> ||
+        std::same_as<std::string_view, record_element_t<field::id, record_t>>) { return std::true_type{}; }),
+      "Requirements for the field-type of the ID-field not met. See documentation for bio::seq_io::reader_options.");
```

Comment:
hmm this looks ok-ish for sequence but will be huge for sam/bam and I remember structure files to be even worse. (We had those concept checks in seqan3). In seqan3 some structure file concepts even had to be outcommented because the compiler just took ages.

I'd propose (since you expect users to mostly use the predefined variables anyway) to provide some "checker" concept that can be applied to reader options nad gives you nice error messages but not having this checked every time. E.g.
`validate_field_types<decltype(my_options)>`.

Reply:
> In seqan3 some structure file concepts even had to be outcommented because the compiler just took ages.

A nice thing is that we can hide the entire block of static asserts with a macro if we do find out that they take too long to compile. It doesn't change any interfaces and is entirely unrelated to overload resolution. It could be `#ifndef NDEBUG` (always included in debug builds) or some explicit `#if BIO_VERBOSE_CONSTRAINTS`.
I can also try adding explicit checks for our types which might improve compile-times. In any case, it is something we should measure and which we can easily add later on.

> Side note:The nice thing in seqan3 was that certain field types could not be set. In bio I can currently set just any type, even the ones like mapq that have a clear type. Can we do something about this?

I thought about this in the beginning, but all of these attempts would make the mechanism less general and would introduce more "special" behaviour that needs extra testing and documentation. I think that the mechanism is quite simple now ("these are all the field types") and would prefer not to add extra rules ("the list of types is the list of fields minus 2 because two fields are not customisable and are automaticall subtracted").